src/context.rs

```diff
@@ -351,15 +351,22 @@ impl<'buf> Secp256k1<AllPreallocated<'buf>> {
 
     /// Creates a context from a raw context.
     ///
+    /// The returned [`core::mem::ManuallyDrop`] context will never deallocate the memory pointed to
+    /// by `raw_ctx` nor destroy the context. This may lead to memory leaks. `ManuallyDrop::drop`
+    /// (or [`core::ptr::drop_in_place`]) will only destroy the context; the caller is required to
+    /// free the memory.
+    ///
     /// # Safety
-    /// This is highly unsafe, due to the number of conditions that aren't checked.
-    /// * `raw_ctx` needs to be a valid Secp256k1 context pointer.
-    /// that was generated by *exactly* the same code/version of the libsecp256k1 used here.
-    /// * The capabilities (All/SignOnly/VerifyOnly) of the context *must* match the flags passed to libsecp256k1
-    /// when generating the context.
-    /// * The user must handle the freeing of the context(using the correct functions) by himself.
-    /// * Violating these may lead to Undefined Behavior.
     ///
+    /// This is highly unsafe due to a number of conditions that aren't checked, specifically:
+    ///
+    /// * `raw_ctx` must be a valid pointer (live, aligned...) to memory that was initialized by
+    ///   `secp256k1_context_preallocated_create` (either called directly or from this library by
+    ///   one of the context creation methods - all of which call it internally).
+    /// * The version of `libsecp256k1` used to create `raw_ctx` must be the same as the version
+    ///   used by this library.
```

Comment:
Is it about version (leading people to think "version number") or should it be the exact same binary? Maybe compiling with different flags could break it?

Reply:
It should be the exact same binary ... I agree that "same version" understates the requirement here. But given that there are no "version numbers" for libsecp, at least not until yesterday, I don't think there's any practical way to interpret this dangerously.

Reply:
I'm pretty sure I've seen version numbers whether they are "real" doesn't matter that much since people will misinterpret it.

Reply:
Okay, let's change this text to clarify that the version used to create `raw_ctx` must be **exactly the one linked into this library**.

Reply:
Used as suggested and force pushed, no other changes.